src/anyio/_core/_exceptions.py

```diff
@@ -65,12 +66,20 @@ def __str__(self):
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}({self.message!r}, {self.errors!r})'
 
+    @staticmethod
+    def check_direct_subclass(exc: BaseException, parents: Tuple[Type[BaseException]]) -> bool:
+        for cls in exc.__class__.mro()[:-1]:
```

Comment:
https://docs.python.org/3/library/inspect.html#inspect.getmro ?

Reply:
Would that work differently here in some scenario?

Reply:
https://docs.python.org/3/library/stdtypes.html?highlight=mro#class.mro yep, the mro method is for overriding the `__mro__` in a metaclass - not for getting the current `__mro__`